pkg/ingress/model_build_target_group.go

```diff
@@ -74,21 +75,31 @@ func (t *defaultModelBuildTask) buildTargetGroupBindingSpec(ctx context.Context,
 	}
 }
 
-func (t *defaultModelBuildTask) buildTargetGroupBindingNetworking(_ context.Context) *elbv2model.TargetGroupBindingNetworking {
-	if t.managedSG == nil {
+func (t *defaultModelBuildTask) buildTargetGroupBindingNetworking(ctx context.Context) *elbv2model.TargetGroupBindingNetworking {
+	if !t.manageBackendSecurityGroupRules {
```

Comment:
see comments above, i think the logic can be simplified if we only have a single 
t.backendSGIDTokens variable captured in the modelBuilder